agent/src/jmh/java/com/code_intelligence/jazzer/runtime/FuzzerCallbacksOptimizedNonCritical.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.runtime;
+
+import com.github.fmeum.rules_jni.RulesJni;
+import java.io.UnsupportedEncodingException;
+
+public final class FuzzerCallbacksOptimizedNonCritical {
+  static {
+    RulesJni.loadLibrary("fuzzer_callbacks", FuzzerCallbacksOptimizedNonCritical.class);
+  }
+
+  static native void traceSwitch(long val, long[] cases, int pc);
+
+  static native void traceMemcmp(byte[] b1, byte[] b2, int result, int pc);
```

Comment:
I think it's the same order, it's just that some methods are missing: There is no optimized non-critical version of `traceCmpInt` (it only takes primitives) and no critical non-Java `traceStrstr` since strings don't benefit from `JavaCritical`.